packages/abi-typegen/src/utils/getMandatoryInputs.ts

```diff
@@ -0,0 +1,40 @@
+import { clone } from 'ramda';
+
+import { EmptyType } from '../abi/types/EmptyType';
+import { OptionType } from '../abi/types/OptionType';
+import type { IType } from '../types/interfaces/IType';
+import type { JsonAbiArgument } from '../types/interfaces/JsonAbi';
+
+import { findType } from './findType';
+
+export type ArgumentWithMetadata<TArg extends JsonAbiArgument = JsonAbiArgument> = TArg & {
+  isOptional: boolean;
```

Comment:
A `getMandatoryInputs` returning an `isOptional` field is counterintuitive. I think `getFunctionInputs` is a better name.

Reply:
I was planning on changing - just didn't have a good name yet.

Reply:
b27050190e2a155cc104b248374b3e9fbe600fec